src/flp.rs

```diff
@@ -545,6 +554,23 @@ pub trait Type: Sized + Eq + Clone + Debug {
     }
 }
 
+/// A type which supports adding noise to aggregate shares for Server Differential Privacy.
+#[cfg(feature = "experimental")]
+pub trait TypeWithNoise<S>: Type
+where
+    S: DifferentialPrivacyStrategy,
+{
+    /// Optionally add noise to the aggregate share to obtain differential privacy.
+    /// Post-condition: The size of `_aggregate_share` has not changed.
+    fn add_noise_to_agg_share<R: Rng>(
```

Comment:
nit: At the level of FLP there is no notion of "aggregate share". Instead we have a concept of a "result" (as in aggregate result). See for example `Type::deocde_result`.
```suggestion
    fn add_noise_to_result<R: Rng>(
```